src/Hastings/Database/Game.hs

```diff
@@ -0,0 +1,129 @@
+                  -> ByteString -- ^The password of the game
+                  -> IO ()
+setPasswordOnGame uuid password = runDB $
+  Esql.update $ \games -> do
+    Esql.set games [GamePassword Esql.=. Esql.val password]
+    Esql.where_ (games Esql.^. GameUuid Esql.==. Esql.val uuid)
+
+-- |Set the name of the game
+setNameOnGame :: String -- ^The UUID of the game
+              -> String -- ^The new name of the game
+              -> IO ()
+setNameOnGame uuid name = runDB $
+  Esql.update $ \games -> do
+    Esql.set games [GameName Esql.=. Esql.val name]
+    Esql.where_ (games Esql.^. GameUuid Esql.==. Esql.val uuid)
+
+-- |Set the number of players in a game
+setNumberOfPlayersInGame :: String -- ^The UUID of the game.
+                         -> Int    -- ^The new number of players.
+                         -> IO ()
+setNumberOfPlayersInGame uuid numberOfPlayers = runDB $
+  Esql.update $ \games -> do
+      Esql.set games [GameMaxAmountOfPlayers Esql.=. Esql.val numberOfPlayers]
+      Esql.where_ (games Esql.^. GameUuid Esql.==. Esql.val uuid)
+
+-- |Set the owner of a game.
+setGameOwner :: Esql.Key Game -- ^The key of the game.
+             -> SessionID     -- ^The sessionID of the new owner.
+             -> IO ()
+setGameOwner gameKey sid = runDB $
+  Esql.update $ \games -> do
+    Esql.set games [GameOwner Esql.=. Esql.val sid]
+    Esql.where_ (games Esql.^. GameId Esql.==. Esql.val gameKey)
+
+-- |Add a player to a game
+addPlayerToGame :: SessionID -- ^The sessionID of the player.
+                -> Esql.Key Game -- ^The key of the game.
+                -> IO (Esql.Key PlayerInGame)
+addPlayerToGame sessionID gameKey = runDB $ Esql.insert $ PlayerInGame gameKey sessionID
+
+-- |Remove a player from a game
+removePlayerFromGame :: SessionID -- ^The sessionID od the player.
+                     -> Esql.Key Game -- ^The key of the game.
+                     -> IO ()
+removePlayerFromGame sessionID gameKey = runDB $
+  Esql.delete $ Esql.from $ \playersInGame ->
+    Esql.where_ (playersInGame Esql.^. PlayerInGameGame Esql.==. Esql.val gameKey
+        Esql.&&. playersInGame Esql.^. PlayerInGamePlayer Esql.==. Esql.val sessionID)
+
+-- |Retrieves a players that are currently in a game.
+retrievePlayersInGame :: Esql.Key Game -- ^The key of the game.
+                      -> IO [Esql.Entity Player]
+retrievePlayersInGame gameKey = runDB $
+      Esql.select $ Esql.from $ \players -> do
+        (playerInGame, onlinePlayers) <- onlinePlayerInGame
+        Esql.where_ (onlinePlayers Esql.^. OnlinePlayerPlayer Esql.==. players Esql.^. PlayerId
+            Esql.&&. playerInGame Esql.^. PlayerInGameGame Esql.==. Esql.val gameKey)
+        return players
+  where
+    onlinePlayerInGame = Esql.from $ \(playerInGame `Esql.InnerJoin` onlinePlayers) -> do
+        Esql.on (playerInGame Esql.^. PlayerInGamePlayer Esql.==. onlinePlayers Esql.^. OnlinePlayerSessionID)
+        return (playerInGame, onlinePlayers)
```

Comment:
Please comment this function (inside it and outside it) more thoroughly. Same can be said about all of these database functions to increase understandability :fireworks: